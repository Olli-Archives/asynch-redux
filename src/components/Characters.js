import React from 'react';
import PropTypes from 'prop-types';
import Character from './Character';
export default function Characters({ characters }){

  const characterList = characters.map((character, i)=>{
    <li key={i}><Character character={character}/></li>;
  });

  return (
    <ul>
      {characterList}
    </ul>
  );
}

Characters.propTypes = {
  characters:PropTypes.array.isRequired
}
;
