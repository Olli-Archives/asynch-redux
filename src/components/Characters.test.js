import React from 'react';
import Characters from './Characters';
import { shallow } from 'enzyme';

describe('Characters component', ()=>{

  it('renders all characters with a character array', ()=>{

    const characterArray = [
      {

        'name': 'Rick Sanchez',
        'image': 'https://rickandmortyapi.com/api/character/avatar/1.jpeg',
      },
      {
        'name': 'Morty Smith',
        'image': 'https://rickandmortyapi.com/api/character/avatar/2.jpeg',
      }
    ];



    const wrapper = shallow(<Characters characters={characterArray}/>);
    expect(wrapper).toMatchSnapshot();
  });
})
;
