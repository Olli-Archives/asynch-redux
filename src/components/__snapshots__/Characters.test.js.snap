// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Characters component renders all characters with a character array 1`] = `
<ul>
  <li
    key="0"
  >
    <Character
      image="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
      name="Rick Sanchez"
    />
  </li>
  <li
    key="1"
  >
    <Character
      image="https://rickandmortyapi.com/api/character/avatar/2.jpeg"
      name="Morty Smith"
    />
  </li>
</ul>
`;
